shapeshift/web
# fix: populate qr code correctly

## Description



i broke this on the send receive multi account branch playing whack a mole fixing a dangerous bug where switching an asset would not change the to address, which could potentially be on the wrong chain.

the `useEffect` to clear the input field on asset change was not the correct way to do this.

we now clear the `Input` field when we handle the asset selection.

this causes the validation handler on the input field to run and clear the hidden/internal `Address` and
`VanityAddress` fields to be reset.

additionally, this moves the trim logic from the input `onChange` handler into the validation function such that all the biz logic is in one place, avoid a react error about changing an input element from an uncontrolled to controlled
component.

## Notice




## Pull Request Type


## Issue (if applicable)



closes #2739

## Risk



* this could have broken sending for any or all assets

## Testing



* open the send modal
* scan a qr code
* go all the way through to the confirm screen - ensure the `Send To` address is populated correctly
* go back to the send details page (where you type amounts)
* change the asset to something different
* it will take you back to the address input page which should be cleared with no console errors, and the `Next` button should be disabled

### Engineering



### Operations



## Screenshots (if applicable)

n/a - no visual changes, qr scanner should work again
